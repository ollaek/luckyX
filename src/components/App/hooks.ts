import { useDispatch, useSelector } from "react-redux"

import { TGlobalState } from "../..";
import { getTopStoresAsync, getTopCategoriesAsync } from "../../actions";
import { TAppState, TopStoresRequestModel } from "../../types";

const useAppState = () => {
    const state = useSelector<TGlobalState, TAppState>(({ appModule }) => appModule);
    const dispatch = useDispatch();
    return {
        ...state,
        getTopStores: (req:TopStoresRequestModel) => getTopStoresAsync(dispatch,req),
        getTopCategories: () => getTopCategoriesAsync(dispatch)
    };
};

export {
    useAppState
};