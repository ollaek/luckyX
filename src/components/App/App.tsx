import React, { useEffect, useState } from "react";

import AdsToStores from "../AdsToStores/AdsToStores";
import Header from "../Header/Header";
import Footer from "../shared/Footer/Footer";
import Loader from "../shared/Loader/Loader";
import SignInModal from "../SignIn/SignInModal";
import TopCashbackCategories from "../TopCashbacksCategories/TopCashbacksCategories";
import TopCashbackStores from "../TopCashbacksStores/TopCashbacksStores";
import { useAppState } from "./hooks";
import { useConfigState } from "../shared/configHook";

const App = ({ history }) => {
  const { configs, getConfig } = useConfigState();
  const {
    isLoading,
    topStores,
    topCategories,
    getTopStores,
    getTopCategories
  } = useAppState();
  const [show, setShow] = useState(false);
  const token = localStorage.getItem("Token");

  useEffect(
    () => {
      if (!token) {
        setTimeout(() => {
          setShow(true);
        }, 1000);
      }
      getConfig();
    },
    // eslint-disable-next-line
    []
  );
  useEffect(
    () => {
     
      if (configs) {
        getTopStores({
          LanguageId: configs.LanguageId,
          Featured: true,
          FeaturedMerchantsNumber: configs.FeaturedMerchantsNumber,
          pageSize: 8,
          pageIndex: 0
        });
        getTopCategories();
      }
    },
    // eslint-disable-next-line
    [configs]
  );
  return (
    <div>
      <Header history={history} />
      <SignInModal history={history} show={show} setShow={val => setShow(val)} />
      <div>
        {topStores && (
          <TopCashbackStores stores={topStores.AffiliateMerchantsList} />
        )}
        {isLoading ? (
          <Loader />
        ) : (
          <TopCashbackCategories categories={topCategories} />
        )}
        <AdsToStores />
      </div>
      {isLoading && <Loader />}
      <Footer />
    </div>
  );
};

export default App;
