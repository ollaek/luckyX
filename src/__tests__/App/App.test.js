import React from "react";
import { shallow, mount } from "enzyme";
import App from "../../components/App/App";
import { Provider } from "react-redux";
import { combineReducers, createStore } from "redux";
import {
  appReducer,
  authReducer,
  onlineCashbacksReducer,
  inStoreOffersReducer
} from "../../reducers";

const rootReducer = combineReducers({
  appModule: appReducer,
  authModule: authReducer,
  onlineCashbacksModule: onlineCashbacksReducer,
  inStoreOffersModule: inStoreOffersReducer
});

const store = createStore(rootReducer);

describe("Home Page Component", () => {
  it("Render App Component", () => {
    const wraper = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(wraper).toMatchSnapshot();
  });
//   it("Render App Component with children", () => {
//     const x = mount(
//       <Provider store={store}>
//         <App />
//       </Provider>
//     );
//     expect(x).toMatchSnapshot();
//   });
});
