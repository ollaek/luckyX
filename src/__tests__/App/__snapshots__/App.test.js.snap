// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Component Render App Component 1`] = `ShallowWrapper {}`;
